jigar/shop/spec/shopcart_test.rb

```diff
@@ -0,0 +1,122 @@
+require '/home/dell/Documents/ruby/shop/shopcart'
+
+describe "Shop_cart" do
+    before do
+        $stock = {}
+        $cart = {}
+    end
+
+    describe "Testing of add_product_to_stock => " do
+        it "it should add the new product in the stock" do
+            expect_any_instance_of(Kernel).to receive(:gets).and_return("Oneplus buds\n", "12.4 mm driver size\n", "1500.00\n", "20\n")

```

Comment:
Passing The functions  parameter would better than making those function get user input